src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarEntry.cs

```diff
@@ -468,8 +463,8 @@ private void CreateNonRegularFile(string filePath, string? linkTargetPath)
             }
         }
 
-        // Verifies if the specified paths make sense for the current type of entry.
-        private void VerifyPathsForEntryType(string filePath, string? linkTargetPath, bool overwrite)
+        // Verifies there's a writable destination.
+        private static void VerifyDestinationPath(string filePath, bool overwrite)
```

Comment:
These checks are removed from this function:

- The `targetDirectoryPath` checks the Directory of the target exists when the entry is extracted. There is no reason we must require the directory to exist. And, the directory could still be created later.
- The checks when `!string.IsNullOrEmpty(linkTargetPath)` for `HardLink` check things that will cause the hard link creation to fail, so there's no need to verify them.
- `string.IsNullOrEmpty(linkTargetPath)` cannot happen when this function is called.

Reply:
Thanks for the change, and sorry for the lack of answer in the initial issue.

> And, the directory could still be created later.

Do you mind adding tests that extract links pointing to non-existent files and directories? For the TarEntry and the TarFile extract methods, sync and async.

Reply:
Actually, please hold on. I'm currently looking at the code's history. There are many security protections we decided to add, which restrict the functionality of TarFile.ExtractToDirectory and TarEntry.ExtractToFile by design.

Reply:
> The targetDirectoryPath checks the directory of the target exists when the entry is extracted. There is no reason we must require the directory to exist. And, the directory could still be created later.

I've added a test for this.